### ISSUE TITLE
emu-docker networking fails with urlfetch.UrlfetchException: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:877)

### ISSUE THREAD

**Issue by dmarjanovic** [google/android-emulator-container-scripts#159]
1.
I'm trying to make `emu-docker` script running on vanilla CentOS VM with enabled nested virtualization but `emu-docker` fails to do any networking which blocks me early in the process. For some reason `urlfetch` python library doesn't like our networking configuration (or proxy setup) though `curl`/`wget` work just fine. It's the same with both python2/3. Any idea where to look for workaround/fix or how to diagnose underlying issue is welcome.

```
(venv) $ emu-docker list
Traceback (most recent call last):
  File "/home/mmmm/Code/google/android-emulator-container-scripts/venv/lib64/python3.6/site-packages/urlfetch-1.2.1-py3.6.egg/urlfetch.py", line 755, in request
    conn.request(method, request_url, data, reqheaders)
  File "/usr/lib64/python3.6/http/client.py", line 1254, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib64/python3.6/http/client.py", line 1300, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.6/http/client.py", line 1249, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.6/http/client.py", line 1036, in _send_output
    self.send(msg)
  File "/usr/lib64/python3.6/http/client.py", line 974, in send
    self.connect()
  File "/usr/lib64/python3.6/http/client.py", line 1415, in connect
    server_hostname=server_hostname)
  File "/usr/lib64/python3.6/ssl.py", line 365, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib64/python3.6/ssl.py", line 773, in __init__
    self.do_handshake()
  File "/usr/lib64/python3.6/ssl.py", line 1033, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib64/python3.6/ssl.py", line 645, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:877)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mmmm/Code/google/android-emulator-container-scripts/venv/bin/emu-docker", line 33, in <module>
    sys.exit(load_entry_point('emu-docker', 'console_scripts', 'emu-docker')())
  File "/home/mmmm/Code/google/android-emulator-container-scripts/emu/emu_docker.py", line 271, in main
    args.func(args)
  File "/home/mmmm/Code/google/android-emulator-container-scripts/emu/emu_docker.py", line 36, in list_images
    emu_downloads_menu.list_all_downloads(args.arm)
  File "/home/mmmm/Code/google/android-emulator-container-scripts/emu/emu_downloads_menu.py", line 411, in list_all_downloads
    img_infos = get_images_info(arm)
  File "/home/mmmm/Code/google/android-emulator-container-scripts/emu/emu_downloads_menu.py", line 323, in get_images_info
    response = urlfetch.get(url)
  File "/home/mmmm/Code/google/android-emulator-container-scripts/venv/lib64/python3.6/site-packages/urlfetch-1.2.1-py3.6.egg/urlfetch.py", line 760, in request
    raise UrlfetchException(e)
urlfetch.UrlfetchException: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:877)
```

The same error I get when running code in python interpreter with HTTPS (and even after debugging in python I couldn't get more descriptive error about what went wrong; note HTTP works just fine):
```
>>> import urlfetch
>>> urlfetch.get('https://dl.google.com/')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mmmm/.local/lib/python2.7/site-packages/urlfetch.py", line 760, in request
    raise UrlfetchException(e)
urlfetch.UrlfetchException: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:618)
```

With other `python` libs it works as expected, they can fetch content from dl.google.com normally:
```
>>> import urllib
>>> f = urllib.FancyURLopener()
>>> f.retrieve("https://dl.google.com", "test.html")
```
or:
```
>>> import urllib2
>>> response = urllib2.urlopen('https://dl.google.com/')
>>> html = response.read()
```

2.
Is there some pre-check script for `emu-docker` script to tell me if I'm missing any python library on my environment (e.g. something for SSL or some dependency of `urlfetch`)? `source ./configure.sh` doesn't complain about anything.

/cc @pokowaka

**Comment by pokowaka** [comment]
It sounds like this is an issue with the urlfetch library. @ifduyue is the maintainer of this library and might have an idea of why this is happening.

In the meantime let me see if we can use the standard python libs.

**Comment by dmarjanovic** [comment]
Ok, there is a fix in https://github.com/ifduyue/urlfetch/issues/48 on its way already. I guess we can close the issue here. Thank you for the support!

**Comment by dmarjanovic** [comment]
Fixed in `urlfetch` 1.2.2.
